savant/deepstream/nvinfer/processor.py

```diff
@@ -0,0 +1,614 @@
+                    if nvds_obj_meta.class_id == obj.class_id:
+                        obj_cls_id = MERGED_CLASSES[self._element_name].get(
+                            obj.class_id
+                        )
+                        if obj_cls_id is not None:
+                            self._logger_trace(
+                                'Updating regular obj id %s -> %s, label "%s".',
+                                obj.class_id,
+                                obj_cls_id,
+                                obj.label,
+                            )
+                            nvds_obj_meta.class_id = obj_cls_id
+                        nvds_set_obj_selection_type(
+                            obj_meta=nvds_obj_meta,
+                            selection_type=ObjectSelectionType.REGULAR_BBOX,
+                        )
+                        try:
+                            nvds_set_obj_uid(
+                                frame_meta=nvds_frame_meta,
+                                obj_meta=nvds_obj_meta,
+                            )
+                        except UIDError:
+                            pass
+                        nvds_obj_meta.obj_label = build_model_object_key(
+                            self._element_name, obj.label
+                        )
+        self._restore_frame(buffer)
+
+    def _process_regular_classifier_output(self, buffer: Gst.Buffer):
+        """Processes output of nvinfer classifier."""
+        self._logger.debug(
+            'Preparing "%s" element regular classifier output, buffer PTS %s.',
+            self._element_name,
+            buffer.pts,
+        )
+        self._model: NvInferAttributeModel
+        nvds_batch_meta = pyds.gst_buffer_get_nvds_batch_meta(hash(buffer))
+        for nvds_frame_meta in nvds_frame_meta_iterator(nvds_batch_meta):
+            for nvds_obj_meta in nvds_obj_meta_iterator(nvds_frame_meta):
+                self._restore_object_meta(nvds_obj_meta)
+                for nvds_clf_meta in nvds_clf_meta_iterator(nvds_obj_meta):
+                    if nvds_clf_meta.unique_component_id != self._model_uid:
+                        continue
+
+                    for attr, label_info in zip(
+                        self._model.output.attributes,
+                        nvds_label_info_iterator(nvds_clf_meta),
+                    ):
+                        nvds_add_attr_meta_to_obj(
+                            frame_meta=nvds_frame_meta,
+                            obj_meta=nvds_obj_meta,
+                            element_name=self._element_name,
+                            name=attr.name,
+                            value=label_info.result_label,
+                            confidence=label_info.result_prob,
+                        )
+        self._restore_frame(buffer)
+
+    def _logger_trace(self, msg: str, *args, **kwargs):
+        if self._logger.isEnabledFor(logging.TRACE):
+            self._logger.trace(msg, args, **kwargs)
```

Comment:
This seems to go against the stated goal of removing extra calls. I suggest not wrapping the logger calls into another function. 

If saving a few lines of code is important, I'd say it is better to remove the `isEnabledFor` condition check entirely, since we do not actually have any expensive args to evaluate in those log messages.